src/routes/memberships.rs

```diff
@@ -64,32 +62,25 @@ pub async fn create(
 
 pub async fn delete(
     conn: &mut Conn,
-    (current_user, db): (User, Db),
+    (actor, db): (PermissionsActor, Db),
 ) -> Result<impl Handler, Error> {
-    let membership_id = conn.param("membership_id").unwrap();
-    let membership_id = Uuid::parse_str(membership_id).map_err(|_| Error::NotFound)?;
+    let membership_id = conn
+        .param("membership_id")
+        .unwrap()
+        .parse::<Uuid>()
+        .map_err(|_| Error::NotFound)?;
 
-    let (membership, account) = Memberships::find_by_id(membership_id)
-        .find_also_related(Accounts)
+    let membership = Memberships::find_by_id(membership_id)
         .one(&db)
         .await?
         .ok_or(Error::NotFound)?;
 
-    let account = account.ok_or(Error::NotFound)?;
-
-    if membership.user_email == current_user.email {
-        return Err(Error::AccessDenied);
-    }
-
-    if !current_user.is_admin() {
-        account
-            .find_related(Memberships)
-            .filter(MembershipColumn::UserEmail.eq(&current_user.email))
-            .one(&db)
-            .await?
-            .ok_or(Error::NotFound)?;
+    if matches!(&actor, PermissionsActor::User(user, _) if membership.user_email == user.email) {
+        Err(Error::AccessDenied)
```

Comment:
Clarification. In English: a user cannot revoke their own membership?

Reply:
Correct. This was a simple way of making sure we had no accounts with zero members that have access. Requiring someone else to remove a member ensures that there's always someone who has active credentials for the account. Otherwise, someone might add a bogus / incorrect email address and remove themselves, making the account inaccessible without divvi up admin intervention.